### ISSUE TITLE
Type definitions not being generated 

### ISSUE THREAD

**Issue by dallasjohnson** [CoinageCrypto/lamington#68]
When trying to compile a project with multiple contracts a `<contract_name>.ts` file created for each contract but all the files only have `// ====================` in them and nothing else.

I have also tried the same with a new project created through lamington init and got the same result.

**Comment by thekevinbrown** [comment]
Interesting, thanks for persisting through these, as you can tell we're still in early days! I'll see if I can repro (and fix) this and will get back to you.

**Comment by dallasjohnson** [comment]
I also wouldn't mind trying to contribute myself and see if I can add some value but I have not had much experience with developing on node. If you can share some advice for how to set this up for local development so I can run this from a local branch rather than just installing from NPM (I guess you must be doing this yourself) then I think there might be quite a few people willing to contribute. I think the Eos ecosystem needs testing tools like this.

**Comment by thekevinbrown** [comment]
No worries at all, that's something that should be documented in general anyway, so I'll get to work on that.

**Comment by MitchPierias** [comment]
In regards to other contributions, can we also lock master to require at least one other dev approval before pull requests are executed

**Comment by dallasjohnson** [comment]
I would not trust myself to push any changes straight into master without a PR and code review from you guys. I don't know enough about NodeJS development. To begin with I would like to be able to run Lamington locally (from a checkout out fork) then I could drop some `console.log()` throughout the code as check point to help diagnose why it's not working as expected on my dev machine. If I was lucky enough to be able to fix something I would like to open a PR for you to review and accept at your discretion.

**Comment by thekevinbrown** [comment]
@MitchPierias that's done!

**Comment by thekevinbrown** [comment]
@dallasjohnson, there's now a guide on how to contribute changes here: https://lamington.io/guides/contributing-to-lamington/

Let me know if you have any trouble doing so.

Otherwise if there's anything we can do to help troubleshoot please let me know. As mentioned on Slack we're having trouble replicating this.

**Comment by dallasjohnson** [comment]
Great, Thank you for the guide. That will be helpful and I have already made a few small tweaks which I will share via PR soon. I now have it all working for me after initially tracking my problem down to an apparent issue with `fs.write` which would only write the first line for me of the type definitions. I had fixed this by joining the lines in memory before then writing to the file with one write command.
Embarrassingly the issue turned out to be much more simple. My local copy of Node which I had installed via brew was not working as expected. The brew install command had worked but my PATH was picking up a very old version of Node which apparently almost worked except for something to do with `fs.write`. If anyone else if silly enough to have the similar issue to me make sure if you have run `brew install node` that the version you get out when you run is `node -v` match up. If not a bit of googling will help fix it or check out this article (this is for Fish shell but I have seen similar issues for Bash) https://medium.com/@joshuacrass/nvm-on-mac-for-fish-users-e00af124c540 <https://medium.com/@joshuacrass/nvm-on-mac-for-fish-users-e00af124c540> 

All is fixed now and I’m starting to make progress updating the tests for my Contracts. 

Thank you for your hard work on this.
